src/hotspot/share/gc/g1/g1FullCollector.cpp

```diff
@@ -225,15 +228,20 @@ void G1FullCollector::complete_collection() {
   _heap->print_heap_after_full_collection(scope()->heap_transition());
 }
 
-void G1FullCollector::update_attribute_table(HeapRegion* hr) {
-  if (hr->is_free()) {
+void G1FullCollector::update_attribute_table(HeapRegion* hr, bool force_pinned) {
+  if (force_pinned) {
+    // Pin high live ratio region
+    _region_attr_table.set_pinned(hr->hrm_index());
     return;
   }
+
   if (hr->is_closed_archive()) {
     _region_attr_table.set_closed_archive(hr->hrm_index());
   } else if (hr->is_pinned()) {
     _region_attr_table.set_pinned(hr->hrm_index());
   } else {
+    // Update _region_attr_table after free pinned regions,
+    // as the region can not be accessed in G1ResetPinnedClosure.
```

Comment:
Not sure what the comment wants to indicate. What does "after free pinned regions" mean here?
If we need a comment I think something like "Everything else is processed normally." should be fine.
But maybe this comment wants to make aware of something special, it's just I do not understand what it would mean.

Reply:
This is a change in behavior to make sure we don't reset humongous regions that have been freed. I'm not sure we need this comment here. Setting the state to `normal` seems ok if we don't want a special state for `free`.